htmresearch/frameworks/nlp/imbu.py

```diff
@@ -145,6 +147,11 @@ def _modelFactory(self, modelType, **kwargs):
                     prepData=False,
                     retinaScaling=1.0)
 
+    elif modelType == "Keywords":
+      # k should be the number of data samples (i.e. numLabels) because Keywords
```

Comment:
No, k should (at a minimum) be the number of data samples. For example, consider a query "paternity". The classifier will give you [the first k matches](https://github.com/numenta/nupic/blob/master/src/nupic/algorithms/KNNClassifier.py#L652) that have the word "paternity". So if k=7 but the number of data samples that contain "paternity" is 10, we'll mistakenly ignore three TPs.